lib/src/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,503 @@
+                .filter(e => e.selected)
+                .map(e => {
+                    const item = this.list.filter((el: Item) => el.key === e.value)[0];
+                    return item;
+                });
+            this.nativeOnChange && this.nativeOnChange(event);
+            this.selectedValue = items;
+        }
+    }
+
+    /** Function which handles the logic of setting the non-native onChange prop (and sets the internal selected value as well) */
+    handleOnChange(value: Item | Array<Item>): void {
+        this.nativeOnChange && this.nativeOnChange(value);
+        this.selectedValue = value;
+    }
+
+    /** Function containing the clear button logic */
+    handleClear(): void {
+        this.handleOnChange(null);
+        this.open = false;
+    }
+
+    /** The native onchange event function that runs when the search input value changes */
+    handleOnChangeSearch(event: any): void {
+        if (this.currentFocused !== -1) {
+            this.currentFocused = -1;
+        }
+        this.searchText = event.target.value;
+    }
+
+    /** Function containing the select dropdown item logic */
+    optionItemSelected(item: Item): void {
+        if (!this.multi) {
+            const newItem = { ...item };
+            this.handleOnChange(newItem);
+            this.open = false;
+        } else {
+            const currentList = (this.selectedValue as Array<Item>) ? (this.selectedValue as Array<Item>) : [];
+            const index = currentList.findIndex((e: Item) => e.key === item.key);
+            if (index === -1) {
+                const newItem = { ...item };
+                const newList = [...currentList, newItem];
+                this.handleOnChange(newList);
+            } else {
+                const newList = currentList.filter((e: Item) => e.key !== item.key);
+                this.handleOnChange(newList);
+            }
+        }
+        this.handleFocus();
+    }
+
+    /** The native event function that runs when the dropdown button is clicked */
+    handleClickToggle(event: any): void {
+        if (!this.disabled) {
+            this.open = !this.open;
+        }
+    }
+
+    /** Function containing the select all button logic */
+    handleSelectAll(): void {
+        if (this.allSelected) {
+            this.handleOnChange([]);
+        } else {
+            this.handleOnChange(this.list);
+        }
+        this.handleFocus();
+    }
+
+    // HELPERS ================================
+    /** Returns the appropriate title for different situations and component types */
+    getTitleLabel() {
+        if (this.uniqueList && this.uniqueList.length === 0) {
+            return "Empty";
+        }
+        if (this.selectedList && this.selectedList.length > 0) {
+            if (this.allSelected) {
+                return `All selected (${this.selectedList.length})`;
+            }
+            if (this.multi) {
```

Comment:
```suggestion
            else if (this.multi) {
```

Reply:
I prefer to have separate if statements when I have a return inside it. It's more readable - just like a switch statement would be.

Reply:
I don't think its a matter of preference though. If the conditions are dependent meaning one occurring negate the other, Then it should be an if else . This is equivalent to having multiple switch cases for a related condition. I disagree with this.

Reply:
"If the conditions are dependent meaning one occurring negate the other"
The fact that different conditions negate one another is true in the case of an if/else statements and switch statement. Usually a switch statements would go through different values of the same variable so it's not very useful here. 

As for why I am not using `else if` I already explained that when an if statement terminates the function because of the return statement inside I prefer to have separate if statements. 

`else if` makes is seem like the else if statements and the rest of the code will still get evaluated which they won't in this case. 

Honestly it doesn't make much difference one way or another. If you don't agree maybe we can decide as a team which style we prefer for this use case in the future.

Reply:
@omarVodiaK @kherP @yousifalraheem ?